src/vhost_user/master.rs

```diff
@@ -452,6 +458,43 @@ impl VhostUserMaster for Master {
         Ok(())
     }
 
+    fn get_inflight_fd(&mut self) -> Result<(VhostUserInflight, RawFd)> {
+        let mut node = self.node();
+        if node.acked_protocol_features & VhostUserProtocolFeatures::INFLIGHT_SHMFD.bits() == 0 {
+            return error_code(VhostUserError::InvalidOperation);
+        }
+
+        let hdr = node.send_request_header(MasterReq::GET_INFLIGHT_FD, None)?;
+        let (inflight, fds) = node.recv_reply_with_fds::<VhostUserInflight>(&hdr)?;
+
+        if let Some(fds) = fds {
+            if fds.len() == 1 {
+                return Ok((inflight, fds[0]));
+            }
+        }
+
+        error_code(VhostUserError::IncorrectFds)
+    }
+
+    fn set_inflight_fd(&mut self, inflight: &VhostUserInflight, fd: RawFd) -> Result<()> {
+        let mut node = self.node();
+        if node.acked_protocol_features & VhostUserProtocolFeatures::INFLIGHT_SHMFD.bits() == 0 {
+            return error_code(VhostUserError::InvalidOperation);
+        }
+
+        if inflight.mmap_size == 0
+            || inflight.mmap_offset >= inflight.mmap_size
```

Comment:
We should not check mmap_offset >= mmap_size, according to the spec:
```
mmap size: a 64-bit size of area to track inflight I/O
mmap offset: a 64-bit offset of this area from the start of the supplied file descriptor
```

Reply:
You're right, I misread the spec, the mmap offset relates to the fd.